src/ilthermoml/dataset.py

```diff
@@ -47,11 +51,26 @@ def __post_init__(self, dataset: Dataset | None) -> None:
 
             raise EntryError(msg) from e
 
-        if len(ilt_entry.components) > 1:
+        if len(components := ilt_entry.components) > 1:
             msg = "entries with multiple components are not supported"
 
             raise EntryError(msg)
 
+        if not (ionic_liquid_smiles := components[0].smiles):
+            msg = "could not retrieve ionic liquid SMILES from entry {self.id!r}"
+
+            raise EntryError(msg)
+
+        try:
+            self.ionic_liquid = IonicLiquid(ionic_liquid_smiles, id=self.id)
```

Comment:
```suggestion
            self.ionic_liquid = IonicLiquid(ionic_liquid_smiles, id=ilt_entry.components[0].id)
```

`self.id` is id of the entry.

Reply:
@chinise 708d0261c07498f9ddd3744aa7a8d2a760dce69a